src/pages/appform.tsx

```diff
@@ -0,0 +1,143 @@
+        console.log(error);
+      });
+
+  return (
+    <div className='grid grid-cols-12 text-xl'>
+      <div className='flex flex-col col-span-10 col-start-2 md:col-span-4 md:col-start-5'>
+        <span className='text-3xl font-light my-2 self-center'>
+          Job Application
+        </span>
+        <form
+          name='job_application'
+          className='flex flex-col shadow-md rounded-lg text-left mt-4'
+          onSubmit={handleSubmit(onSubmit)}
+        >
+          <div className='h-2 bg-purple-400 rounded-t-md'></div>
+          <label htmlFor='company' className='font-semibold ml-4 py-2'>
+            Company
+          </label>
+          {errors.company && 'Company name is required'}
+          <input
+            type='text'
+            placeholder='Enter company here...'
+            className='border px-3 py-1 mx-4 hover:outline-none focus:outline-none focus:ring-indigo-500 focus:ring-1 rounded-md'
+            {...register('company', { required: true })}
+          />
+          <label htmlFor='position' className='font-semibold ml-4 py-2'>
+            Position
+          </label>
+          {errors.position && 'Position is required'}
+          <input
+            type='text'
+            placeholder='Enter positon here...'
+            className='border px-3 py-1 mx-4 hover:outline-none focus:outline-none focus:ring-indigo-500 focus:ring-1 rounded-md'
+            {...register('position', { required: true })}
+          />
+          <label htmlFor='url' className='font-semibold ml-4 py-2'>
+            URL
+          </label>
+          {errors.url && 'url is required'}
+          <input
+            type='text'
+            placeholder='Enter url here...'
+            className='border px-3 py-1 mx-4 hover:outline-none focus:outline-none focus:ring-indigo-500 focus:ring-1 rounded-md'
+            {...register('url', { required: true })}
+          />
+          <label htmlFor='source' className='font-semibold ml-4 py-2'>
+            Source
+          </label>
+          <select
+            className='max-w-[36%] border px-3 py-1 mx-4 hover:outline-none focus:outline-none focus:ring-indigo-500 focus:ring-1 rounded-md'
+            {...register('source')}
+          >
+            <option value='LinkedIn'>LinkedIn</option>
+            <option value='Seek'>Seek</option>
+          </select>
+          <label htmlFor='status' className='font-semibold ml-4 py-2'>
+            Status
+          </label>
+          <select
+            className='max-w-[36%] border px-3 py-1 hover:outline-none focus:outline-none focus:ring-indigo-500 focus:ring-1 rounded-md mx-4'
+            {...register('status')}
+          >
+            <option value='APPLIED'>Applied</option>
+            <option value='REJECTED'>Rejected</option>
+            <option value='PHONE_SCREEN'>Phone Screen</option>
+            <option value='INTERVIEW'>Interview</option>
+            <option value='JOB_OFFER'>Job Offer</option>
```

Comment:
This could simplified to 

```TSX
<select>
    {
        Object.keys(StatusEnum).map((key: string) => {
            <option 
                key={key} 
                value={key}
            >
                {StatusEnum[key]}
            </option>
        })
    }
</select>
```

Reply:
Was giving me type error but this resolved the issue.

https://stackoverflow.com/questions/57086672/element-implicitly-has-an-any-type-because-expression-of-type-string-cant-b
![ss222](https://user-images.githubusercontent.com/6619450/217497201-6ea0af76-4575-4b71-9cbc-05ff2f6df4f7.png)

Reply:
Sorry about that. I've commented in your new PR about how to fix this.

https://github.com/codesydney/job-tracker/pull/28#discussion_r1099953037

That was my fault, I missed extracting the `keyof typeof`  `as const` object and should have let you know.